### PR TITLE
ENH: Raise on attempt to set a lazyval.

### PR DESCRIPTION
**Pull request by ssanderson**
ping @richafrank who noticed this odd behavior.

### CODE REVIEW

**Comment by ssanderson** [comment]
@llllllllll does this seem reasonable?

**Comment by llllllllll** [comment]
I believe this is the correct behavior. It also matches the `val` part of the name. 
```scala
scala> lazy val a = 10
a: Int = <lazy>

scala> a = 20
<console>:11: error: reassignment to val
       a = 20
         ^
```

File: zipline/utils/memoize.py [hunk 1]
```diff
@@ -44,6 +50,9 @@ def __get__(self, instance, owner):
             self._cache[instance] = val = self._get(instance)
             return val
 
+    def __set__(self, instance, owner):
```
Comment (richafrank) [inline zipline/utils/memoize.py]: I realize it's unused here, but should the last parameter be named `value`?

Comment (llllllllll) [inline zipline/utils/memoize.py]: yes

**Comment by richafrank** [comment]
Thanks @ssanderson !

**Comment by ssanderson** [comment]
Updated the name on the `__set__` arg.

**Comment by llllllllll** [comment]
merge on pass

**Pull request merged**